minimal-plugin/src/main/java/io/micronaut/gradle/Dependencies.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.gradle;
+
+import org.gradle.api.plugins.JavaPlugin;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Builder class to model build dependencies.
+ * @author Sergio del Amo
+ * @since 3.4.0
+ */
+public class Dependencies {
+    private static final String DEVELOPMENT_ONLY = "developmentOnly";
+
+    private final List<String> developmentOnlyDependencies;
+    private final List<String> implementationDependencies;
+    private final List<String> testImplementationDependencies;
+    private final List<String> runtimeOnlyDependencies;
+    private final List<String> compileOnlyDependencies;
+
+    private Dependencies(List<String> developmentOnlyDependencies,
+                        List<String> implementationDependencies,
+                        List<String> testImplementationDependencies,
+                         List<String> runtimeOnlyDependencies,
+                         List<String> compileOnlyDependencies) {
+        this.developmentOnlyDependencies = developmentOnlyDependencies;
+        this.implementationDependencies = implementationDependencies;
+        this.testImplementationDependencies = testImplementationDependencies;
+        this.runtimeOnlyDependencies = runtimeOnlyDependencies;
+        this.compileOnlyDependencies = compileOnlyDependencies;
+    }
+
+    public Map<String, List<String>> toMap() {
+        Map<String, List<String>> m = new HashMap<>();
```

Comment:
:x: Actually this should probably be a `LinkedHashMap`. Otherwise we are order sensitive, which, in general, won't be an issue, but by experience, this can cause some difficult to diagnose cache misses, in case dependencies are reordered.

Reply:
Resolved via https://github.com/micronaut-projects/micronaut-gradle-plugin/pull/434/commits/5d48c7869ae3731f401da21736899196fee8921f